routes/libraries.js

```diff
@@ -239,150 +256,180 @@ router.route('/:id/survey')
      * }
      * This is using the error checking from before, using the helper functions for uniformity.
      */
+    // Grab all of the inputs from the request body.
+    let genresInput = [
+      req.body.pictureBooks,
+      req.body.youngAdultFiction,
+      req.body.fantasyFiction,
+      req.body.fairyTale,
+      req.body.boardBook,
+      req.body.nonFiction,
+      req.body.mystery,
+      req.body.graphicNovel,
+      req.body.chapterBooks,
+    ];
+
+    // For every value, if it does not exist, then the checkbox was not selected.
+    genresInput = genresInput.filter((genre) => {
+      return typeof genre === "string";
+    });
+
     try {
       updateData.fullness = validation.isValidNumber(
-        updateData.fullness,
+        parseInt(updateData.fullness),
         "Fullness Rating"
       );
-      updateData.genres = validation.checkStringArray(
-        updateData.genres,
+      genresInput = validation.checkStringArray(
+        genresInput,
         "Genres Available"
       );
+      if (genresInput.length === 0 && updateData.fullness !== 0) {
+        throw "Must specify at least one genre for a non-empty library.";
+      }
     } catch (e) {
-      return res.status(400).json({ error: e });
+      return res
```

Comment:
Why would this be a 500 error?

Reply:
Didn't see this comment until later, sorry that it's outdated.

Didn't know what else to put for now, the code can be changed. I put anything I didn't know what to do as a 500 because it seemed the most vague to spit back if the form was submitted incorrectly (from an external request not from the form), and I wanted to put something that worked with the error page.

Atm it just puts "Internal Server Error" for poorly done data, and now re-renders on failed logic cases (such as no genres for full library, genres for an empty library).

Hope that gives a little bit of insight.